jetty-servlets/src/main/java/org/eclipse/jetty/servlets/IncludeExcludeBasedFilter.java

```diff
@@ -122,10 +122,28 @@ protected boolean shouldFilter(HttpServletRequest http_request, HttpServletRespo
 
         String content_type = http_response.getContentType();
         LOG.debug("Content Type is: {}",content_type);
-        content_type = (content_type == null)?"":content_type;
-        String mime_type = MimeTypes.getContentTypeWithoutCharset(content_type);
 
-        LOG.debug("Mime Type is: {}",content_type);
+        String mime_type = "";
+        if (content_type != null)
+        {
+            mime_type = MimeTypes.getContentTypeWithoutCharset(content_type);
+            LOG.debug("Mime Type is: {}",mime_type);
+        }
+        else
+        {
+            String request_url = http_request.getPathInfo();
+            String filename = request_url.substring(request_url.lastIndexOf("/") + 1,request_url.length());
```

Comment:
It is not necessary to strip the leading path elements. getDefaultMimeByExtension will handle them OK